src/HttpMethodsClient.php

```diff
@@ -79,12 +92,55 @@ public function options($uri, array $headers = [], $body = null): ResponseInterf
 
     public function send(string $method, $uri, array $headers = [], $body = null): ResponseInterface
     {
-        return $this->sendRequest($this->requestFactory->createRequest(
-            $method,
-            $uri,
-            $headers,
-            $body
-        ));
+        if (!is_string($uri) && !$uri instanceof UriInterface) {
```

Comment:
Note that this is not a BC break, even if the caller didn't have strict typing and expected stuff to be cast to a string at runtime, because this file itself has strict typing and was previously forwarding the parameter as is, so the result would still be a type error later, but just a more obscure error message. Including an early guard here will help callers correct their error, and also checking the uri and body are the correct types here mean our logic later on within the private createRequest function is a little simpler than it would have otherwise been.